lib/logstash/outputs/nagios.rb

```diff
@@ -105,14 +103,24 @@ def receive(event)
     @logger.debug("Opening nagios command file", :commandfile => @commandfile,
                   :nagios_command => cmd)
     begin
-      File.open(@commandfile, "r+") do |f|
-        f.puts(cmd)
-        f.flush # TODO(sissel): probably don't need this.
-      end
+      send_to_nagios(cmd)
     rescue => e
       @logger.warn("Skipping nagios output; error writing to command file",
                    :commandfile => @commandfile, :missed_event => event,
                    :exception => e, :backtrace => e.backtrace)
     end
   end # def receive
-end # class LogStash::Outputs::Nagios
+
+  private
+
+  def commandfile?
+    File.exists?(@commandfile)
+  end
```

Comment:
I would rename this to `command_file_exist?`, using `commandfile?` make be believe that we check if the commandfile return true, which isn't the case.